components/style/properties/longhand/font.mako.rs

```diff
@@ -1813,22 +1820,31 @@ ${helpers.single_keyword_system("font-variant-position",
                     return Err(())
                 }
 
+                let b_tag = tag.deref().as_bytes();
+                let mut buff: [u8;4] = [0;4];
+
+                for i in 0..4 {
```

Comment:
You could just zip here like `buff.iter_mut().zip(b_tag.as_bytes())` which will return once the lengths mismatch.